File: code/js/src/domain/Delivery.ts
export interface DeliveryDomain {
    id: number,
    dueDate: Date,
    tagControl: string,
    assignmentId: number,
    lastSync: Date,
}

export class DeliveryDomain {
    constructor(
        id: number,
        dueDate: Date,
        tagControl: string,
        assignmentId: number,
        lastSync: Date,
    ) {
        this.id = id
        this.dueDate = dueDate
        this.tagControl = tagControl
        this.assignmentId = assignmentId
        this.lastSync = lastSync
    }
}

File: code/jvm/main/kotlin/com/isel/leic/ps/ionClassCode/domain/Delivery.kt
package com.isel.leic.ps.ionClassCode.domain

import java.sql.Timestamp

/**
 * Delivery Domain Interface
 */
data class Delivery(
    val id: Int,
    val dueDate: Timestamp,
    val tagControl: String,
    val assignmentId: Int,
    val lastSync: Timestamp,
)
